charts/core/src/Chart/hooks/useChart.ts

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useRef, useState } from 'react';
+import { LineChart as EchartsLineChart } from 'echarts/charts';
+import {
+  GridComponent,
+  LegendComponent,
+  TitleComponent,
+  ToolboxComponent,
+  TooltipComponent,
+} from 'echarts/components';
+import * as echarts from 'echarts/core';
+import { CanvasRenderer } from 'echarts/renderers';
+import debounce from 'lodash.debounce';
+
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { DarkModeProps } from '@leafygreen-ui/lib';
+
+import { ChartOptions, SeriesOption } from '../../Chart/Chart.types';
+import { colors } from '../colors';
+import { getDefaultChartOptions } from '../config';
+
+import { addSeries, removeSeries, updateOptions } from './updateUtils';
+
+/**
+ * Register the required components. By using separate imports, we can avoid
+ * importing the entire echarts library which will reduce the bundle size.
+ * Must be added to if additional functionality is supported.
+ */
+echarts.use([
+  TitleComponent,
+  TooltipComponent,
+  GridComponent,
+  LegendComponent,
+  EchartsLineChart,
+  CanvasRenderer,
+  ToolboxComponent,
+]);
+
+interface ChartHookProps extends DarkModeProps {
+  onChartReady?: () => void;
+}
+
+/**
+ * Creates a generic Apache ECharts options object with default values for those not set
+ * that are in line with the designs and needs of the design system.
+ */
+export function useChart({
+  darkMode: darkModeProp,
+  onChartReady,
+}: ChartHookProps) {
+  const chartRef = useRef(null);
+  const chartInstanceRef = useRef<echarts.EChartsType | undefined>();
+  const { theme } = useDarkMode(darkModeProp);
+  const [chartOptions, setChartOptions] = useState(
+    getDefaultChartOptions(theme),
+  );
+
+  useEffect(() => {
+    const chartInstance = echarts.init(chartRef.current);
+    chartInstanceRef.current = chartInstance;
+    chartInstance.setOption(chartOptions);
+
+    // ECharts does not automatically resize when the window resizes.
+    const resizeHandler = () => {
+      chartInstance.resize();
+    };
+    window.addEventListener('resize', resizeHandler);
+
+    if (onChartReady) {
+      chartInstance.on('finished', onChartReady);
+    }
+
+    return () => {
+      window.removeEventListener('resize', resizeHandler);
+      chartInstance.dispose();
+    };
+  }, []);
+
+  const updateChartRef = debounce((chartOptions: Partial<ChartOptions>) => {
+    chartInstanceRef.current?.setOption(chartOptions);
+  }, 50);
```

Comment:
Still not convinced we need the debounce. I feel like there's a way around this

Reply:
Definitely open to ideas here. Just running a small test, I added a log in this method to see how many updates run on a chart with 100 lines, with and without debouncing the update call. 

With `debounce` it just runs once and there is 1 `onChartReady` action triggered.
<img width="126" alt="Screenshot 2024-10-29 at 10 07 09 AM" src="https://github.com/user-attachments/assets/46b9fe16-53d9-4126-951b-a0a0c397c947">
<img width="476" alt="Screenshot 2024-10-29 at 10 07 15 AM" src="https://github.com/user-attachments/assets/423601ee-00a5-4321-a38c-71f6f4f74d21">

Without `debounce` however, due to re-renders, the chart is update 400+ times and triggering 400+ `onChartReady` calls as well:
<img width="168" alt="Screenshot 2024-10-29 at 10 06 26 AM" src="https://github.com/user-attachments/assets/64210f2b-0b64-46ba-ade2-6b6bc0e19996">
<img width="486" alt="Screenshot 2024-10-29 at 10 06 40 AM" src="https://github.com/user-attachments/assets/cdc4220a-861c-4ae0-9e84-5c5ccc2601fb">

So if we didn't want to debounce the call, I still think something is required here to fix this.